tonilopezmr/day4/PassphraseValidatorPartTwo.kt

```diff
@@ -0,0 +1,108 @@
+package day4
+
+
+import org.junit.Assert.*
+import org.junit.Test
+import java.io.File
+import java.util.*
+
+class PassphraseValidatorPartTwo {
+
+  fun validI(passphrase: String): Boolean {
+    val words = passphrase
+        .split(" ")
+    return words
+        .foldIndexed(true) { idx, acc, a ->
+          acc && words.subList(idx + 1, words.size)
+              .fold(true) { bcc, b ->
+                bcc && !equalsInAnyOrder(a, b)
+              }
+        }
+  }
+
+  fun containsInAnyOrder(passphrase: List<String>, element: String): Boolean =
+          passphrase.fold(false) { bcc, b ->
+            if (equalsInAnyOrder(element, b)) true
+            else bcc
+          }
+
+  fun equalsInAnyOrder(a: String, b: String): Boolean = a.fold(true) { acc, it ->
+    if (a.length != b.length) false else acc && b.contains(it)
+  }
```

Comment:
This is okay but you can check if you strings are anagrams if you sort them and they are equal. That will simplify your solution a lot 👍

Reply:
Yes but I don't what is better for performance? good thought

Reply:
I don't know what will be the cost of `sort` integrated method. But right now you have 4 nested loops. @JoseLlorensRipolles and me only have 2, I think, and we are using this approach.